packages/message/src/Granules.ts

```diff
@@ -246,15 +246,15 @@ export const generateGranuleApiRecord = async ({
   }
 
   // FUTURE: null files are currently not supported in update payloads
-  // RDS Phase 3 should revise logic to accept an explict null value
+  // RDS Phase 3 should revise logic to accept an explicit null value
   if (files === null) {
     throw new CumulusMessageError('granule.files must not be null');
   }
 
   const {
     granuleId,
     cmrLink,
-    published = false,
+    published,
```

Comment:
So the idea here is to set the defaults as far up as possible, yeah? I think that's a good call having this logic in the endpoint and message write logic.

It makes me a little nervous that we have `writeGranuleFromApi` and it's callers `createGranuleFromApi` + `updateGranuleFromApi` available as exported functions that don't set the defaults correctly but... that may be OK. They don't appear to be used in a way that would cause problems now, only in the endpoints you've updated. In the future if someone were to call e.g. `writeGranuleFromApi` directly without a `published` field, there might be problems so we'd have to know not to do that.

Reply:
> It makes me a little nervous that we have writeGranuleFromApi and it's callers createGranuleFromApi + updateGranuleFromApi available as exported functions that don't set the defaults correctly but... that may be OK.

That's not entirely unfair - if we feel that those methods should defend against the defaults being set in non patch situations we could write a wrapper that throws (?).  

That said, I think we want the write logic chain  to know as little about business-logic defaults as possible, particularly in P3, as the database *schema* should be handling fall-through validation concerns, and I think we should be calling what's in the api model for API record validation before getting to the db write phase.  

Meaning the business of setting defaults/incoming validation should be in the input/output endpoint logic and/or message write logic, with the write then being forced to conform to the schema in the postgres DB by virtue of the write, *generally*.        It's possible to also set defaults within the db schema, although that has issues with the current logic and replication to dynamo, and is inconsistent with the approach.

Reply:
Agreed to punt to an arch meeting/P3.